### PR TITLE
RBN-97: Fetching book reviews

### PR DESCRIPTION
**Pull request by isidora-b**
### What did you implement?

Implemented review repository, service and controller for fetching all reviews of a book, added unit tests and integrations tests.

### How can we verify the changes?

Changes can be verified by running the app and checking "/books/{bookId}/reviews" endpoint in Swagger.

### More info

- Jira: [RBN-97](https://productdock.atlassian.net/browse/RBN-97)

### CODE REVIEW

File: rbc/src/main/java/com/library/rbc/exceptionHandler/BookNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.library.rbc.exceptionHandler;
```
Comment (tamastarjan) [inline rbc/src/main/java/com/library/rbc/exceptionHandler/BookNotFoundException.java]: Package name should be lowercase.

File: rbc/src/main/java/com/library/rbc/exceptionHandler/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+  @ExceptionHandler(value = BookNotFoundException.class)
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  public @ResponseBody ErrorResponse handleException(BookNotFoundException ex) {
+    return new ErrorResponse(HttpStatus.NOT_FOUND.value(), ex.getMessage());
+  }
+}
```
Comment (tamastarjan) [inline rbc/src/main/java/com/library/rbc/exceptionHandler/GlobalExceptionHandler.java]: The status code is included in the status line of the HTTP response, so there is no need to also include it in the response body, i.e. in the `ErrorResponse` class.

File: rbc/src/test/java/com/library/rbc/controller/reviewcontroller/ReviewControllerShould.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  void getResponseWhenNoBookIsFound() {
+    Pageable pageable = PageRequest.of(PAGE_NUMBER, PAGE_SIZE);
+    when(reviewService.getReviewsByBookId(BOOK_ID, pageable)).thenThrow(
+        new BookNotFoundException("There is no book with id: " + BOOK_ID));
+
+    BookNotFoundException exception = assertThrows(BookNotFoundException.class, () -> {
+      reviewController.getReviewsByBookId(BOOK_ID, pageable);
+    });
+
+    String expectedMessage = "There is no book with id: " + BOOK_ID;
```
Comment (tamastarjan) [inline rbc/src/test/java/com/library/rbc/controller/reviewcontroller/ReviewControllerShould.java]: Phrasing suggestion: `Book with ID 'some-id' was not found.`

**Review by tamastarjan — APPROVED** [review]
Review: LGTM.

**Review by Aleksandra-Romanic — APPROVED** [review]

**Pull request merged**